edit_readme

ryo-kamido opened the pull request:
# What
DB設計を行った。
また、categoryテーブルは階層構造であるため、parent_idを使用。
# Why
サービスに必須機能かつ、サービスの根幹となる機能のため。

以下、各テーブルの役割について記載する。

# userテーブル
ユーザー情報の管理を行うテーブル。
パスワードとメールアドレスをユニーク化させることで、重複ユーザーの登録を防止。
また、Facebook Gmail Twitterカラムには、Facebookからのログインをしている場合は1を入れる、などにして管理。
どのユーザーがFacebookなどのサービスからログインしているか、がわかるようにカラムを作成した。

# itemテーブル
商品情報を管理するテーブル。
statusではitem状態のステータス(出品中など)を管理
item_statusではitemの商品としての状態(新品、中古など)を管理

# categoryテーブル
商品カテゴリーを管理するテーブル
カテゴリーが階層形式になっているため、parrent_idを使うことで階層を表現する。
また、パンくずリストなども作成することからparrent_idを使うことが良いのではないか、と考えた。
(各階層ごとのテーブルを作成することも検討したが、アソシエーションが困難だったため)

# item_commentテーブル
購入前、取引成立前の掲示板のコメントを管理するテーブル

# tradeテーブル
取引情報について管理するテーブル。
取引の状態(取引終了など)を表すstatusや、
出品者　購入者評価を管理するparcaser_evaluation　saler_evaluationをカラムとして実装。
また、購入者　出品者の取引が途中で中止になった場合はこのtradeテーブルのidを消すようにすれば取引そのものがなかったことに出来る。

# saler_userテーブル
出品者のデータを管理するテーブル
仮売上金temporary_sales_moneyをカラムとして設定。
取引が正式に締結した場合に、この仮売上金がuserテーブルのsales_moneyカラムに追加されていくように実装する予定。
仮売上金は、取引内容によって手数料10%以外にも配送料などで変容する可能性があるため、実装した。

# percaser_userテーブル
購入者のデータを管理するテーブル

# trade_commentテーブル
契約した購入者と出品者の二人のみが利用できる掲示板のコメントを管理するテーブル。

# regionテーブル
あらかじめ47都道府県のデータを入れておくテーブル
(外部から追加することはなく、あくまでも運営側があらかじめ用意しているデータを渡すためのテーブル)

# brandテーブル
あらかじめブランドのデータを入れておくテーブル
(外部から追加することはなく、あくまでも運営側があらかじめ用意しているデータを渡すためのテーブル)

Commit:
edit_readme

#What
DB設計を行った。
また、categoryテーブルは階層構造であるため、parent_idを使用。
#Why
サービスに必須機能かつ、サービスの根幹となる機能のため。

TOMARUMARU reviewed (commented):
usersテーブルに情報が多い気がするので、別途profileモデルを用意してあげたほうが良さそうです。

prefectureに関してはactive hashというものを使用することをオススメしているので、調べてみてください。

Commit:
edit(2)_ReadMe

# WHAT
reviewを受けて、READMEを編集した。
Profileテーブルを作成し、Userテーブルにはログインに必要な情報(passwordとemail-address)のみを残した。
また、regionテーブルを削除。
prefectureカラムを作成し、active_hashを実装する予定。

# Why
USERテーブルの情報が多すぎたため。

Commit:
edit(3)_README

belongs_toのスペルミスを直した

ryo-kamido commented:
> usersテーブルに情報が多い気がするので、別途profileモデルを用意してあげたほうが良さそうです。
> 
> prefectureに関してはactive hashというものを使用することをオススメしているので、調べてみてください。

修正しました。

miyagit commented on README.md:
```diff
@@ -1,24 +1,160 @@
+|nickname|string|null: false|
+|birthdate|DATE|null: false|
+|postal_code|integer|null: false|
+|prefecture|string|null: false|
```
prefectureはいらないですね。active_hashで実装するので。

ryo-kamido replied:
それ以外の修正箇所はございませんでしょうか？
ご確認のほど、よろしくお願い致します。

ReoKuriyama replied:
追加でコメントしました

ryo-kamido replied:
確認いたしました

ReoKuriyama commented on README.md:
```diff
@@ -1,24 +1,160 @@
+
+|Column|Type|Options|
+|------|----|-------|
+|user_id|integer|null: false,foreign_key: true|
```
profile_idの間違いですかね？

ryo-kamido replied:
修正いたしました

ReoKuriyama commented on README.md:
```diff
@@ -1,24 +1,160 @@
+|sales_money|integer|null: false|
+|point|integer|null: false|
+|payment_methods|string|null: false|
+|Facebook|integer|
```
token等が入るカラムでしょうか？
カラム名分かりづらいので見直してください

ryo-kamido replied:
こちらのカラムは、Facebookユーザーか否かを判断するためのカラムでしたが、
trelloの指示に「has_many sns_credentials的なテーブルを作成して、uid, providerを保存できるようにする(必要ならtokenも)」とあったため、今後の実装で追加することにいたします。

Commit:
edit(4)_README

# WHAT
1.
Pointカラムの名前をわかりやすく、mercari-pointカラムとした。
2.
Facebookカラムなどを削除
今後の実装でテーブルを追加する際に、
Facebookカラムなどに関わる情報が保管されているテーブルを作成予定
3.
user_idとなっている箇所をprofile_idに変更

ReoKuriyama reviewed (approved):
LGTMです 👍

ryo-kamido committed:
Merge branch 'master' into edit_ReadMe
